_posts/2016-10-19-katello-puppet4-integration.md

```diff
@@ -0,0 +1,132 @@
+
+You can use this matrix as a quick reference of what can work together:
+
+| | Puppet 3 client | Puppet 4 client
+| :--- | :---: | :---: 
+| **Puppet 4 server** | Compatible | Compatible
+| **Puppet 3 server** | Compatible | Not compatible
+
+Katello will by default install modules to the normal file paths for the version of Puppet you are using. 
+Puppet 3 will install modules to `/etc/puppet/environments`
+Puppet 4 will install modules to `/etc/puppetlabs/code/environments`
+
+### How to upgrade both Katello servers, capsules, and hosts to use Puppet 4
+
+We have a [great video](https://youtu.be/GFNPHypFhl4?t=2545) and [excellent documentation](http://www.katello.org/docs/nightly/upgrade/puppet.html) on how to upgrade from Puppet 3 to 4.
+
+To upgrade an existing Katello instance to Puppet 4:
+
+Before starting, we recommend creating a backup or a snapshot of your system.
+
+Then run:
+
+```bash
+katello-service stop # This will stop all services.
+foreman-installer --upgrade-puppet # This performs the upgrade.
+```
+
+The --upgrade-puppet command will restart the services, but you can check that they are running with `hammer ping`. If you run puppet --version on the server, it should show `4.x.x`. 
+
+### How to install Katello with Puppet 4
+
+#### Production Environments
+If you want to create a new Katello server configured with Puppet 4, the easiest way is to spin up a Virtual Machine with [Forklift](https://github.com/Katello/forklift).
+
+To do this you can follow these instructions:
+
+```bash
+git clone https://github.com/katello/forklift.git
+cd forklift
+vagrant up centos7-katello-p4
+```
+
+This will setup a virtual environment with Katello and Puppet 4 enabled.
+
+
+#### Development environment
+
+If you are so inclined, you can create a development environment of Katello using Puppet 4. To do this, spin it up (also with Forklift) as a virtual machine with the following configuration:
+
+```yaml
+example-dev-p4:
+  box: centos7
+  memory: 9000
+  shell: 'yum -y install ruby && cd /vagrant && ./setup.rb'
+  options: --scenario=katello-devel --puppet-four
+  installer: --katello-devel-github-username <YOUR GITHUB USERNAME> --disable-system-checks
+  ansible:
+    group: 'server'
+```
+* Note that this setup does require you to have a fork of Katello setup on your Github account.
+
+Then `vagrant up example-dev-p4` and let it spin up the box.
+
+After this, you'll need to do a few things to get it running.
+
+
+
+#### Making Capsules
+
+If you wanted to create a capsule with Forklift, you can follow these steps:
+
+* In your boxes.yml file, you can write:
+```yaml
+example-capsule-p4:
+  box: centos7
+  ansible:
+    playbook: 'playbooks/capsule-dev.yml'
+    options: --puppet-four
+    group: 'capsule'
+    server: 'example-katello-p4'
+```
```

Comment:
Note: If https://github.com/Katello/forklift/pull/279 is merged, we should change this to ansible